pyfolio/perf_attrib.py

```diff
@@ -118,21 +130,83 @@ def create_perf_attrib_stats(perf_attrib):
         empyrical.sharpe_ratio(specific_returns)
 
     summary['Cumulative specific returns'] =\
-        empyrical.cum_returns(specific_returns)
+        empyrical.cum_returns_final(specific_returns)
     summary['Cumulative common returns'] =\
-        empyrical.cum_returns(common_returns)
+        empyrical.cum_returns_final(common_returns)
     summary['Total returns'] =\
-        empyrical.cum_returns(perf_attrib['total_returns'])
+        empyrical.cum_returns_final(perf_attrib['total_returns'])
 
     summary = pd.Series(summary)
     return summary
 
 
-def show_perf_attrib_stats(perf_attrib_data, risk_exposures):
+def show_perf_attrib_stats(returns, positions, factor_returns,
+                           factor_loadings):
     """
-    Takes perf attribution data over a period of time, computes stats on it,
-    and displays them using `utils.print_table`.
+    Calls `perf_attrib` using inputs, and displays outputs using
+    `utils.print_table`.
     """
+    risk_exposures, perf_attrib_data = perf_attrib(returns,
+                                                   positions,
+                                                   factor_returns,
+                                                   factor_loadings)
+
     perf_attrib_stats = create_perf_attrib_stats(perf_attrib_data)
     print_table(perf_attrib_stats)
     print_table(risk_exposures)
+
+
+def plot_returns(returns, specific_returns, common_returns, ax=None):
+    """
+    Plot total, specific, and common returns.
+    """
+    if ax is None:
+        ax = plt.gca()
+
+    ax.plot(returns, color='g', label='Total returns')
+    ax.plot(specific_returns, color='b', label='Cumulative specific returns')
+    ax.plot(common_returns, color='r', label='Cumulative common returns')
+
+    ax.set_title('Time Series of Cumulative Returns')
+    ax.set_ylabel('Returns')
+    ax.legend()
+
+    return ax
+
+
+def plot_alpha_returns(alpha_returns, ax=None):
```

Comment:
It's important to `return ax` in any function that plots stuff.

Each of pyfolio's tear sheets should be one single image: that is, if I saved it the whole tear sheet should save as a single png, not one png for each plot in the tear sheet. This is to make it easier for the analysis to be saved, passed around, etc. We go to great lengths in `tears.py` to achieve this, and the way we do it is by making a single `matplotlib` figure with several axes, and pass these axes to functions in other modules: `tears.py` then expects the axes back, with the stuff having been plotted.

We also don't lose the "just plot" functionality with this method either, because if a user just calls `plot_foo` and _doesn't_ pass in an axes, then `ax=None`, `plt.gca()` and the `ax` that gets returned is just thrown away, leaving only the plotted output to be shown with `plt.show()`. So we don't really lose anything here.

There are some other small necessary changes that I'll outline in separate comments. Hope this made sense!